docs: update gov spec with proposal changes (#10594)
cosmos/cosmos-sdk: x/gov/spec/01_concepts.md

- **Vote:** Once deposit reaches a certain value (`MinDeposit`), proposal is
  confirmed and vote opens. Bonded Atom holders can then send `TxGovVote`
  transactions to vote on the proposal.
- **Execution** After a period of time, the votes are tallied and depending
  on the result, the messages in the proposal will be executed.

## Proposal submission

### Right to submit a proposal

> Every account, can submit proposals by sending a `MsgSubmitProposal` transaction.

Reviewer: ```suggestion
Every account can submit proposals by sending a `MsgSubmitProposal` transaction.
```

Once a proposal is submitted, it is identified by its unique `proposalID`.

### Proposal Messages

A proposal includes an array of `sdk.Msg`s which are executed automatically if the

> proposal passes. The messages are executed by the governance module itself. Modules

Reviewer: ```suggestion
proposal passes. The messages are executed by the governance `ModuleAccount` itself. Modules
```

such as `x/upgrade`, that want to allow certain messages to be executed by governance
only should add a whitelist within the respective msg server, granting the governance
module the right to execute the message once a quorum has been reached. The governance

> module uses the message router to check that these messages are correctly constructed

Reviewer 2: ```suggestion
module uses the `MsgServiceRouter` to check that these messages are correctly constructed
```
